### ISSUE TITLE
Use CLAW to add OpenACC directive to kernels

### ISSUE THREAD

**Issue by arporter** [stfc/PSyclone#142]
In SRS ticket 322 (https://code.metoffice.gov.uk/trac/lfric/ticket/322) I investigated the steps that we needed to take in order to target OpenACC from PSyclone. One of those steps was the addition of the directive:

    !$acc routine

to each subroutine which we wish to execute on the GPU. In this issue we will use CLAW to perform this task.

**Comment by arporter** [comment]
Depends on #114 (as that provides the CLAW interface).

**Comment by arporter** [comment]
I wanted to write a test for this using the GOcean API but the fact that it still uses `WRITE` and `READ` in kernel meta-data causes the Omni compiler to fall over. There's issue https://github.com/stfc/dl_esm_inf/issues/3 for `dl_esm_inf` and I've created a new issue (#144) to update PSyclone accordingly.

**Comment by arporter** [comment]
The OpenACC spec says that the `routine` directive can appear anywhere within the declarations part of a routine.

**Comment by arporter** [comment]
I now have a working script that inserts a `!$acc routine seq` directive into a kernel routine. I was hoping to be able to use CLAW's `claw.tatsu.directive.common.Directive.generateRoutineDirectives()` function but it appears that that expects to be called on the routine that *calls* other functions that then need the directive adding to them (I've asked @clementval about this on slac). However, we just have the one function (the kernel subroutine) that we want to add the directive to. Therefore, I've temporarily hacked CLAW to make `Directive.addPragmasBefore()` public and then called that from my Python script. A better solution would be to implement what I want in CLAW itself (and then it could also be used from `Directive.generateRoutineDirectives()`.

**Comment by arporter** [comment]
Given that all we want to do is add `!$acc routine seq`, this is easily achieved by modifying the kernel AST. Therefore I'm closing this issue for the time being.
